libfuse/errors.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2017 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package libfuse
+
+import (
+	"syscall"
+
+	"bazil.org/fuse"
+	"github.com/keybase/kbfs/kbfsblock"
+	"github.com/keybase/kbfs/kbfsmd"
+	"github.com/keybase/kbfs/libkbfs"
+)
+
+type errorWithErrno struct {
+	error
+	errno syscall.Errno
+}
+
+var _ fuse.ErrorNumber = errorWithErrno{}
+
+func (e errorWithErrno) Errno() fuse.Errno {
+	return fuse.Errno(e.errno)
+}
+
+func filterError(err error) error {
+	switch err.(type) {
```

Comment:
I should actually call `errors.Cause` (or whatever it's called) here

Reply:
Yes please!